rsocket-core/src/main/java/io/rsocket/RSocketServer.java

```diff
@@ -42,8 +49,8 @@
   private final Function<Frame, ? extends Payload> frameDecoder;
   private final Consumer<Throwable> errorConsumer;
 
-  private final NonBlockingHashMapLong<Subscription> sendingSubscriptions;
```

Comment:
I think handleSendProcessorError and handleSendProcessorCancel should now be synchronized as they iterate over sendingSubscriptions/channelProcessors (previously this wasn't necessary because of Cliff Click magic).

Reply:
I wrapped the map with Collections.synchronizedMap so it should effectively be doing that.

Reply:
I looked at the code again. I think we are fine - the problem would be if something adds something to map while we're iterating over it. Looking at the non blocking map the same thing would happen. We should probably add something to the request* methods so that you can't add items to the map when handling a cancel or an error, and then this isn't a problem.